dbfit-java/core/src/main/java/dbfit/util/sql/PreparedStatements.java

```diff
@@ -8,11 +8,9 @@ private static String buildParamList(int numberOfParameters) {
         return join(repeat("?", numberOfParameters), ", ");
     }
 
-    public static String buildStoredProcedureCall(String procName, int numberOfPararameters) {
-        return "{ call " + procName + "(" + buildParamList(numberOfPararameters) + ")}";
-    }
-
-    public static String buildFunctionCall(String procName, int numberOfParameters) {
-        return "{ ? = call " + procName + "(" + buildParamList(numberOfParameters - 1) + ")}";
+    public static String buildStoredRoutineCallCmdText(
+            String routineName, int numberOfPararameters, boolean paramsIncludeReturnValue) {
+        return "{ " + (paramsIncludeReturnValue ? "? = " : "") + "call " + routineName + "(" +
```

Comment:
Yep, it's better. Still a little bit convoluted. I would think of something like following if I have to describe the syntax:
```
routine_invocation: <routine_name>(<params_list>)
routine_with_return_value:    { ? = call <routine_invocation> }
routine_without_return_value: { call <routine_invocation> }
```

That's subjective though... (there could be simpler way to express it). My point is just to make that form more obvious from the code.